### PR TITLE
metview: switch to `openssl@3`

### PR DESCRIPTION
**Pull request by carlocab**
This has linkage with `openssl@1.1`. Spotted in #134638.

### CODE REVIEW

**Comment by iMichka** [comment]
```
2023-06-22T20:18:01.0539506Z [ 63%] Building C object mars-client/src/CMakeFiles/marsclient.dir/account.c.o
2023-06-22T20:18:01.0543409Z cd /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/mars-client/src && /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/gcc-11 -DLITTLE_END -DNOFDB -DNOFDB5 -DNOODB -DR64 -Dmarsclient_EXPORTS -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/mars-client/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/mars-client/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/mir/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/mir/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/eckit/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/eckit/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/atlas/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/atlas/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/atlas/atlas_io/src -I/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/atlas/atlas_io/src -I/usr/include/tirpc -I/home/linuxbrew/.linuxbrew/include/eigen3 -pipe -O3 -DNDEBUG -fPIC -MD -MT mars-client/src/CMakeFiles/marsclient.dir/account.c.o -MF CMakeFiles/marsclient.dir/account.c.o.d -o CMakeFiles/marsclient.dir/account.c.o -c /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/mars-client/src/account.c
2023-06-22T20:18:01.0543876Z In file included from /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/mars-client/src/account.c:15:
2023-06-22T20:18:01.0544385Z /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/mars-client/src/mars.h:39:10: fatal error: rpc/rpc.h: No such file or directory
2023-06-22T20:18:01.0544503Z    39 | #include <rpc/rpc.h>
2023-06-22T20:18:01.0544601Z       |          ^~~~~~~~~~~
2023-06-22T20:18:01.0544722Z compilation terminated.
2023-06-22T20:18:01.0545191Z make[2]: *** [mars-client/src/CMakeFiles/marsclient.dir/build.make:102: mars-client/src/CMakeFiles/marsclient.dir/account.c.o] Error 1
2023-06-22T20:18:01.0545593Z make[2]: Leaving directory '/tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build'
2023-06-22T20:18:01.0545960Z make[1]: *** [CMakeFiles/Makefile2:12424: mars-client/src/CMakeFiles/marsclient.dir/all] Error 2
2023-06-22T20:18:01.0546100Z make[1]: *** Waiting for unfinished jobs....
2023-06-22T20:18:01.0546260Z [ 63%] Linking CXX executable mir_tests_unit_atlas
2023-06-22T20:18:01.0546848Z cd /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/mir/tests/unit && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.4/bin/cmake -E remove EXE_FILENAME-NOTFOUND
2023-06-22T20:18:01.0547540Z cd /tmp/metview-20230622-36665-80inxb/MetviewBundle-2023.4.1-Source/build/mir/tests/unit && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.26.4/bin/cmake -E cmake_link_script CMakeFiles/mir_tests_unit_atlas.dir/link.txt --verbose=1
```

**Comment by cho-m** [comment]
Build is picking up the wrong include directory (`/usr/include/tirpc`) and looks like the `CMAKE_CXX_FLAGS` didn't help when building `mars-client`
```
  -- Found RPC: /home/linuxbrew/.linuxbrew/lib/libtirpc.so  
  -- mars_client FOUND RPC
  --    RPC_INCLUDE_DIRS : [/usr/include/tirpc]
  --    RPC_LIBRARIES : [/home/linuxbrew/.linuxbrew/lib/libtirpc.so]
```

`RPC_PATH=HOMEBREW_PREFIX` probably isn't working as it is missing the `tirpc` subdirectory (`include/tirpc/rpc/rpc.h`):
```cmake
    if( NOT RPC_PATH )
        if ( NOT "$ENV{RPC_PATH}" STREQUAL "" )
            set( RPC_PATH "$ENV{RPC_PATH}" )
        elseif ( NOT "$ENV{RPC_DIR}" STREQUAL "" )
            set( RPC_PATH "$ENV{RPC_DIR}" )
        endif()
    endif()

    find_path(RPC_INCLUDE_DIR NAMES rpc/rpc.h PATHS ${RPC_PATH} ${RPC_PATH}/include /usr/include /usr/include/tirpc NO_DEFAULT_PATH )
```

---

On side note, `-DFFTW_PATH=#{HOMEBREW_PREFIX}` is not being expanded:
https://github.com/Homebrew/homebrew-core/blob/03af85679df71b2cd6842766d3377447d65095f3/Formula/metview.rb#L51-L56

File: Formula/metview.rb [hunk 1]
```diff
@@ -41,21 +43,16 @@ class Metview < Formula
...
+    args = %W[
       -DBUNDLE_SKIP_ECCODES=1
       -DENABLE_MIR_DOWNLOAD_MASKS=OFF
       -DENABLE_BUILD_TOOLS=OFF
       -DENABLE_ECKIT_CMD=OFF
-      -DFFTW_PATH=#{HOMEBREW_PREFIX}
+      -DFFTW_PATH=#{Formula["fftw"].opt_prefix}
     ]
-    system "cmake", "-S", ".", "-B", "build", *args, *cmake_rpc_flags, *std_cmake_args
+    args << "-DRPC_PATH=#{Formula["libtirpc"].opt_prefix}" if OS.linux?
```
Comment (carlocab) [inline Formula/metview.rb:53]: Looks like this didn't work...

File: Formula/metview.rb [hunk 2]
```diff
@@ -41,21 +43,22 @@ class Metview < Formula
...
       -DENABLE_ECKIT_CMD=OFF
-      -DFFTW_PATH=#{HOMEBREW_PREFIX}
+      -DFFTW_PATH=#{Formula["fftw"].opt_prefix}
     ]
-    system "cmake", "-S", ".", "-B", "build", *args, *cmake_rpc_flags, *std_cmake_args
+
+    if OS.linux?
+      args += [
+        "-DRPC_PATH=#{Formula["libtirpc"].opt_prefix}",
+        "-DRPC_INCLUDE_DIRS=#{Formula["libtirpc"].opt_include}/tirpc",
```
Comment (cho-m) [inline Formula/metview.rb:57]: Maybe:
```suggestion
        "-DRPC_INCLUDE_DIR=#{Formula["libtirpc"].opt_include}/tirpc",
```

CMake ends up doing `set( RPC_INCLUDE_DIRS ${RPC_INCLUDE_DIR} )`

Comment (carlocab) [reply Formula/metview.rb]: Nice, that worked. Thanks.

**Review by nandahkrishna — APPROVED** [review]

**Pull request merged**